pkg/azurefile/controllerserver.go

```diff
@@ -1021,7 +1021,7 @@ func (d *Driver) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequ
 	if len(req.SnapshotId) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "Snapshot ID must be provided")
 	}
-	rgName, accountName, fileShareName, _, _, _, err := GetFileShareInfo(req.SnapshotId) //nolint:dogsled
+	rgName, accountName, fileShareName, _, _, subsID, err := GetFileShareInfo(req.SnapshotId) //nolint:dogsled
```

Comment:
unfortunately the snapshot id does not container subs id, e.g. 
```
azurefile-csi-driver-test-201921e8-8897-4f15-9199-d480a5c899fb#f84252b3b42624bd1a4f423#deletesnapshot-volume-1-4da0a823-4cb36f05###default#2025-04-16T13:08:25.0000000Z
```

I think supporting cross subscription for snapshot is a new feature, need to append subs id in creating snapshot

Reply:
no worry, I will fix it

Reply:
@andyzhangx Do you have some estimate on when you'll be able to merge the fix? We're nearing OpenShift 4.19 release and need to decide whether we document a workaround (manual removal) or cherry-pick the fix.

Reply:
changing volumeID format is risky, I will only make such change in csi driver v1.33.0, so I think you need to document a workaround (manual removal)

Reply:
I see, thank you.

Reply:
wait, if sourceVolumeID already contains the subs id, I think my new commit in this PR would fix the issue, it just adds `isValidSubscriptionID` func to validate whether it's subs id or not, that's quite safe change.